pkg/assets/create/creater.go

```diff
@@ -163,36 +163,68 @@ func create(ctx context.Context, manifests map[string]*unstructured.Unstructured
 			continue
 		}
 
+		var resource dynamic.ResourceInterface
 		if mappings.Scope.Name() == meta.RESTScopeNameRoot {
-			_, err = client.Resource(mappings.Resource).Create(manifests[path], metav1.CreateOptions{})
+			resource = client.Resource(mappings.Resource)
 		} else {
-			_, err = client.Resource(mappings.Resource).Namespace(manifests[path].GetNamespace()).Create(manifests[path], metav1.CreateOptions{})
+			resource = client.Resource(mappings.Resource).Namespace(manifests[path].GetNamespace())
 		}
-
 		resourceString := mappings.Resource.Resource + "." + mappings.Resource.Version + "." + mappings.Resource.Group + "/" + manifests[path].GetName() + " -n " + manifests[path].GetNamespace()
 
+		incluster, err := resource.Create(manifests[path], metav1.CreateOptions{})
+
+		if err == nil && options.Verbose {
+			fmt.Fprintf(options.StdErr, "Created %q %s\n", path, resourceString)
+		}
+
 		// Resource already exists means we already succeeded
 		// This should never happen as we remove already created items from the manifest list, unless the resource existed beforehand.
 		if kerrors.IsAlreadyExists(err) {
 			if options.Verbose {
 				fmt.Fprintf(options.StdErr, "Skipped %q %s as it already exists\n", path, resourceString)
 			}
-			delete(manifests, path)
-			continue
+			incluster, err = resource.Get(manifests[path].GetName(), metav1.GetOptions{})
+			if err != nil {
+				if options.Verbose {
+					fmt.Fprintf(options.StdErr, "Failed to get already existing %q %s: %v\n", path, resourceString, err)
+				}
+				errs[path] = fmt.Errorf("failed to get %s: %v", resourceString, err)
+				continue
+			}
 		}
 
 		if err != nil {
 			if options.Verbose {
 				fmt.Fprintf(options.StdErr, "Failed to create %q %s: %v\n", path, resourceString, err)
 			}
-			errs[path] = fmt.Errorf("failed to create: %v", err)
+			errs[path] = fmt.Errorf("failed to create %s: %v", resourceString, err)
 			continue
 		}
 
-		if options.Verbose {
-			fmt.Fprintf(options.StdErr, "Created %q %s\n", path, resourceString)
+		if mstatus, ok := manifests[path].Object["status"]; ok && mstatus != nil {
+			_, found, err := unstructured.NestedMap(incluster.Object, "status")
+			if err != nil {
+				if options.Verbose {
+					fmt.Fprintf(options.StdErr, "Failed to find status field %q %s: %v\n", path, resourceString, err)
+				}
+				errs[path] = fmt.Errorf("failed to find status field %s: %v", resourceString, err)
+				continue
+			}
+			if !found {
+				unstructured.SetNestedMap(incluster.Object, manifests[path].Object["status"].(map[string]interface{}), "status")
```

Comment:
why this complexity with typecasts? Just set it as an untyped value here with`SetNestedField`, and drop the `!= nil` test above.